### PR TITLE
Feature/custom elements

### PR DESCRIPTION
**Pull request by Matsa59**
## 📖 Description

Add a system to extends `ex_prosemirror`

Fixes # (issue)

## 📋 Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### CODE REVIEW

File: assets/js/prosemirror/blocks.js [hunk 1]
```diff
@@ -32,11 +36,14 @@ export default (options) => {
 
   const heading = [];
 
-  jsonOptions.map((option) => {
-    if (allowedHeading[option]) {
-      heading.push(allowedHeading[option]);
-    } else
-      map[option] = blocks[option];
+  blocksSelection.map((/** @type {Object} */ blockSelection) => {
```
Comment (tomkonidas) [inline assets/js/prosemirror/blocks.js:39]: Do we really want the type in the comment like that? My tsserver recommends it too but it looks funny 🤷‍♂️

File: guides/extends_editor.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# Extends the editor
+
+To create a new `mark` / `blocks`, you should first follow prosemirror spec.
```
Comment (tomkonidas) [inline guides/extends_editor.md:3]:
```suggestion
To create a new `mark` / `block`, you should first follow prosemirror spec.
```

File: guides/extends_editor.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+You can found an example [here](https://prosemirror.net/examples/schema/).
```
Comment (tomkonidas) [inline guides/extends_editor.md:5]:
```suggestion
You can find an example [here](https://prosemirror.net/examples/schema/).
```

File: guides/extends_editor.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+## Create a custom span mark
+
+In our example, will have the following mark spec:
```
Comment (tomkonidas) [inline guides/extends_editor.md:9]:
```suggestion
In our example, we will have the following mark spec:
```

File: guides/extends_editor.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+exProsemirror.setCustomMarks({span: spanMark});
+
+exProsemirror.initAll();
+```
+
+
+## Create a custom blocks
+
+In our example, will have the following block spec:
```
Comment (tomkonidas) [inline guides/extends_editor.md:58]:
```suggestion
In our example, we will have the following block spec:
```

**Review by tomkonidas — APPROVED** [review]

**Review by onixus74 — APPROVED** [review]
After comments 👍

File: assets/js/prosemirror/blocks.js (hunk 1 above)
Comment (Matsa59) [reply assets/js/prosemirror/blocks.js:39]: yeah It's look a little bit weird IMO but it define what is the variable so I guess it's not bad to have

**Pull request merged**